src/annotated_commit.h

```diff
@@ -7,11 +7,31 @@
 #ifndef INCLUDE_annotated_commit_h__
 #define INCLUDE_annotated_commit_h__
 
+#include "oidarray.h"
+
 #include "git2/oid.h"
 
-/** Internal structure for merge inputs */
+typedef enum {
+	GIT_ANNOTATED_COMMIT_REAL = 0,
+	GIT_ANNOTATED_COMMIT_VIRTUAL = 1,
+} git_annotated_commit_t;
```

Comment:
I'm wondering whether we should have 0 be an invalid value and have 1 and 2 as real and virtual resp. So we can detect in cleanup functions that we failed to set a value.